### ISSUE TITLE
[Bug] After v4.9.40 update sites sometimes freezes

### ISSUE THREAD

**Issue by realnc** [darkreader/darkreader#7266]
# Bug Report

## Bug Description

I was using Firefox 93.0 before and everything was fine. There was upgrade to Firefox 94.0.1 yesterday. Now some websites stopped working with Dark Reader.

What happens is that Firefox now generated 100% CPU load. The CPU heats up and I can hear the CPU fan speeding up. Also, memory use increases constantly until Dark Reader is stopped. It starts at about 200MB, then 300MB, then 400MB, and so on. If you leave it running, it will keep going into the gigabytes of consumed RAM.

After a while, Firefox shows a message at the top that "Dark Reader is slowing down Firefox".

## Website Address

https://www.phoronix.com/forums/

## Steps To Reproduce

Go to https://www.phoronix.com/forums to trigger the bug.

## System Information

- OS: Gentoo Linux AMD64 X11
- Browser: Firefox 94.0.1
- Dark Reader Version: 4.9.41

**Comment by Gusted** [comment]
Welp, not sure what is going wrong, but my laptop's CPU is really getting hugged.
I was able to bisect it to commit 2c4f9cbe19638a7a25b93022e5c567ae22646a3b.
It's on my To-Do for this weekend.

**Comment by 47an** [comment]
High cpu usage and refuse load https://www.ravenminer.com/blocks with v4.9.40.

OS: Ubuntu 20.04.3
Browser: Version 95.0.4638.69 (Official Build) (64-bit)

**Comment by cipitaua** [comment]
confirmed on ubuntu devel

**Comment by Gusted** [comment]
TL;DR - I've written some very bad code when a much more simpler solution was available and which I did know of, not sure what got me that day. Sorry.

Fixed by #7275

**Comment by Gusted** [comment]
A hot-fix will be pushed out once we've fixed some other _possible_ regressions.

**Comment by baptx** [comment]
This issue is still present, for example on Instagram web version if we check the followers of someone, Firefox says Dark Reader is slowing down. Is there a fix planned? The issue could be reopened for more visibility to contributors.

**Comment by Gusted** [comment]
This is likely caused by another issue, please fire a new issue for this.
